pkg/vz/network_darwin.go

```diff
@@ -50,77 +51,117 @@ func DialQemu(unixSock string) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
+	vzConn := &packetConn{Conn: dgramConn}
 
-	remote := tcpproxy.DialProxy{
-		DialContext: func(context.Context, string, string) (net.Conn, error) {
-			return dgramConn, nil
-		},
-	}
-	go remote.HandleConn(qemuConn)
+	go forwardPackets(qemuConn, vzConn)
 
 	return client, nil
 }
 
-// QEMUPacketConn converts raw network packet to a QEMU supported network packet.
-type QEMUPacketConn struct {
-	unixConn net.Conn
+func forwardPackets(qemuConn *qemuPacketConn, vzConn *packetConn) {
+	defer qemuConn.Close()
+	defer vzConn.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		if _, err := io.Copy(qemuConn, vzConn); err != nil {
+			logrus.Errorf("Failed to forward packets from VZ to VMNET: %s", err)
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		if _, err := io.Copy(vzConn, qemuConn); err != nil {
+			logrus.Errorf("Failed to forward packets from VMNET to VZ: %s", err)
+		}
+	}()
+
+	wg.Wait()
 }
 
-var _ net.Conn = (*QEMUPacketConn)(nil)
```

Comment:
We don't need to verify this since it alway implements the interface now, since we embed a `net.Conn`. Previously we kept a net.Conn instance so the check was useful.